lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -637,6 +637,38 @@ static bool isAsyncCall(const ApplyExpr *call) {
   return funcType->isAsync();
 }
 
+/// Extract the correct expression for diagnosing actor state being passed inout
+///
+/// This unwraps most expression types, working toward an offending DeclRefExpr
+/// or to the LookupExpr before the DeclRef.
+///
+/// The return will either be a LookupExpr or a DeclRefExpr.
+/// If part of the sub-expression isn't interesting, returns nullptr
+static const Expr *getInoutTopExpr(const InOutExpr *arg) {
+
+  const Expr * currentExpr = arg->getSubExpr();
+
+  while (currentExpr) {
+    if (const InOutExpr *inout = dyn_cast<InOutExpr>(currentExpr)) {
+      return nullptr; // The AST walker will hit this again
+    } else if (const LookupExpr *baseArg = dyn_cast<LookupExpr>(currentExpr)) {
+      while (const LookupExpr *nextLayer = dyn_cast<LookupExpr>(baseArg->getBase()))
```

Comment:
I don't think we need this loop, given that you'll go around the outer loop again anyway?

Reply:
Loop gone.